### ISSUE TITLE
[Bug]: Default Post Sort Reverts to 'Hot' on App Close

### ISSUE THREAD

**Issue by artlessdodger** [CreatureSurvive/Arctic-Issues#20]
### Check for open issues

- [X] I have verified that another issue for this is not open, or it has been closed and has not been fixed.

### Minimal reproducible example

In the Settings -> Content menu, I set the Default Post Sort to TopTwelveHour. When I close the app and reopen it, the setting reverts to Hot.

Issue appears to affect both iPhone and iPad. 

### Expected Behavior

I suppose I'd generally expect this setting to stay at TopTwelveHour (or whatever else I picked for the Default Post Sort). Best-case scenario it would also persist at the chosen setting when the app updates.

### Version

v0.4.1.5

### iOS Version

17.x Beta

### Device

iPhone 14 Pro; iPad Pro (11-inch) (4th generation)

**Comment by CreatureSurvive** [comment]
What a coincidence, I received another report about this the other day. I managed to resolve this issue yesterday and it will be included in the next update. 

While I was working on this, I managed to fix a number of other issues regarding the default sort setting. Arctic will never change this setting anymore. In cases where the default is set to (_Scaled_ or_Controversial_) (Lemmy v0.19+ only) Arctic will temporarily revert to _Hot_ sorting if you switch to an account on an instance that is below v0.19.
